### ISSUE TITLE
document behaviour of setting a value with c-variants

### ISSUE THREAD

**Issue by u-fischer** [latex3/latex3#604]
Using c-variants to set a variable sometimes simply silently creates this variable (if one doesn't use check-declarations):

~~~~
\documentclass{article}
%\usepackage[check-declarations]{expl3}
\usepackage{expl3}

\begin{document}
\ExplSyntaxOn
\prop_gput:cnn {g_xxxx_prop}{abc1}{abc} %no error
\prop_gput:Nnn \g_xxxx_prop {abc2}{abc} %no error
\prop_show:N   \g_xxxx_prop %shows the content
\prop_gput:Nnn \g_zzzz_prop {abc}{abc} %error

\ExplSyntaxOff
\end{document}
~~~~

Sometimes one gets low-level errors quite different to the N-variant:

~~~~
\documentclass{article}
%\usepackage[check-declarations]{expl3}
\usepackage{expl3}

\begin{document}
\ExplSyntaxOn
\int_gset:cn{g_zzzz_int}{4}
\ExplSyntaxOff
\end{document}
~~~~

I doubt if one should change the behaviour, but imho it should be at least better documented - I couldn't find anything about it in interface3.

**Comment by FrankMittelbach** [comment]
I agree that there probably should be a discussion on how "c" behaves with so far undeclared variables (with the conclusion that it is best to always declare them first). As you say, there is no really good way to enforce it upfront and there are cases where not declaring them beofrehand can actually be part of the design, but it should be really the exception not the norm.
